pkg/kv/kvserver/replica_rangefeed.go

```diff
@@ -365,7 +365,7 @@ func (r *Replica) registerWithRangefeedRaftMuLocked(
 	r.rangefeedMu.Unlock()
 
 	feedBudget := r.store.GetStoreConfig().RangefeedBudgetFactory.CreateBudget(r.startKey,
-		&r.store.cfg.Settings.SV)
+		MaxCommandSize.Get(&r.store.cfg.Settings.SV), &r.store.cfg.Settings.SV)
```

Comment:
There's a bit of ugliness here with passing settings and also pulling server setting as a separate argument. The reason for doing it is a package structure where raft settings are defined under server and feed settings defined under feed. Currently server depends on feed for the reason of building the server and we can't have a dependency back to read the property from under the budget factory. Is there anything obvious that would allow just passing settings maybe? Can't see an easy way atm.

Reply:
I think passing in an explicit minimum value is better, to decouple these. I think it might even be better for the caller to check `RangefeedBudgetsEnabled` before calling `CreateBudget` rather than calling it inside the method.

Reply:
Drive-by, feel free to ignore, but why aren't we passing the settings to `NewBudgetFactory`? Perhaps then the factory could provide a method that does all of the adjustments, i.e. in `CreateBudget` we'd simply call `factory.budgetSize()` (where zero corresponds to disabled):

https://github.com/cockroachdb/cockroach/blob/c9df00f79c768974c942f14448780159734af0d4/pkg/server/server.go#L484-L485

```go
serverrangefeed.NewBudgetFactory(ctx, kvMemoryMonitor, cfg.MemoryPoolSize, cfg.HistogramWindowInterval(), func() int64 {
  if !serverrangefeed.RangefeedBudgetsEnabled(&cfg.Settings) { return 0 }
 if maxSize := kvserver.MaxCommandSize.Get(&cfg.Settings); maxSize > f.limit { return maxSize }
  return f.limit
})
```

I think this would further need cleanup where we avoid even populating `f.limit`, you get the idea.

Reply:
Weird, I only say @tbg's comment after hours. What I was saying there's another PR #78969 to add a server setting to disable budgets on the fly and it pushes settings to the factory. There's still an issue of not being able to use MaxCommandSize as it is defined inside server without making a cyclic dependency.